File: ShadowFrameworkLite20/src/shadow/geometry/functions/SFSimpleTexCoordGeometryuv.java
package shadow.geometry.functions;

import shadow.math.SFVertex2f;

public class SFSimpleTexCoordGeometryuv extends SFUnoptimizedSurfaceFunctionUV{

	private float du,dv;
	
	public SFSimpleTexCoordGeometryuv(float du, float dv) {
		super();
		this.du = du;
		this.dv = dv;
	}

	@Override
	public SFVertex2f getTexCoord(float u, float v) {
		return new SFVertex2f(u*du, v*dv);
	}

	@Override
	public void init() {
		//Do nothing
	}
	
	@Override
	public void destroy() {
		//Its correct: if init isn't doing anything, destroy should not do anything
	}
	

	protected float getDu() {
		return du;
	}

	protected void setDu(float du) {
		this.du = du;
	}

	protected float getDv() {
		return dv;
	}

	protected void setDv(float dv) {
		this.dv = dv;
	}
}

File: ShadowFrameworkJS/src/shadow/geometry/functions/SFSimpleTexCoordGeometryuv.js


function SFSimpleTexCoordGeometryuv(du,dv){
	this.du=du;
	this.dv=dv;	
}

inherit(SFSimpleTexCoordGeometryuv,SFUnoptimizedSurfaceFunctionUV);


SFSimpleTexCoordGeometryuv.prototype["getTexCoord"]=function(u,v){
	return new SFVertex2f(u*this.du,v*this.dv);
};

SFSimpleTexCoordGeometryuv.prototype["init"]=function(){
	//Do nothing
};

SFSimpleTexCoordGeometryuv.prototype["destroy"]=function(){
	//Do nothing
};	

SFSimpleTexCoordGeometryuv.prototype["getDu"]=function(){
	return this.du;
};	

SFSimpleTexCoordGeometryuv.prototype["getDv"]=function(){
	return this.dv;
};

SFSimpleTexCoordGeometryuv.prototype["setDu"]=function(du){
	this.du=du;
};	

SFSimpleTexCoordGeometryuv.prototype["setDv"]=function(dv){
	this.dv=dv;
};	
